File: src/main/java/com/m2t/hachikodictionary/dto/Response.java
package com.m2t.hachikodictionary.dto;

import java.time.LocalDateTime;

public class Response {
    private LocalDateTime timestamp;
    private Boolean success;
    private String message;
    private Object data;
    private Boolean showNotification;

    public Response(Boolean success, String message) {
        this(LocalDateTime.now(), success, message, null, true);
    }

    public Response(Boolean success, String message, Boolean showNotification) {
        this(LocalDateTime.now(), success, message, null, showNotification);
    }

    public Response(Boolean success, String message, Object data) {
        this(LocalDateTime.now(), success, message, data, true);
    }

    public Response(Boolean success, String message, Object data, Boolean showNotification) {
        this(LocalDateTime.now(), success, message, data, showNotification);
    }

    private Response(LocalDateTime timestamp, Boolean success, String message, Object data, Boolean showNotification) {
        this.timestamp = timestamp;
        this.success = success;
        this.message = message;
        this.data = data;
        this.showNotification = showNotification;
    }

    // Getters
    public LocalDateTime getTimestamp() {
        return timestamp;
    }

    public Boolean getSuccess() {
        return success;
    }

    public String getMessage() {
        return message;
    }

    public Object getData() {
        return data;
    }

    public Boolean getShowNotification() {
        return showNotification;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Response response = (Response) o;

        if (!success.equals(response.success)) return false;
        return message.equals(response.message);
    }

    @Override
    public int hashCode() {
        int result = success.hashCode();
        result = 31 * result + message.hashCode();
        return result;
    }
}

File: src/main/java/com/m2t/hachikodictionary/dto/Response.kt
package com.m2t.hachikodictionary.dto

import java.time.LocalDateTime


data class Response(
    val timestamp: LocalDateTime?,
    val success: Boolean?,
    val message: String?,
    val data: Any? = null,
    val showNotification: Boolean? = true
) {
    constructor(success: Boolean, message: String) : this(
        timestamp = LocalDateTime.now(),
        success = success,
        message = message,
    )

    constructor(success: Boolean, message: String, showNotification: Boolean?) : this(
            timestamp = LocalDateTime.now(),
            success = success,
            message = message,
            showNotification = showNotification
    )

    constructor(success: Boolean, message: String, data: Any) : this(
        timestamp = LocalDateTime.now(),
        success = success,
        message = message,
        data = data
    )

    constructor(success: Boolean, message: String, data: Any, showNotification: Boolean?) : this(
            timestamp = LocalDateTime.now(),
            success = success,
            message = message,
            data = data,
            showNotification = showNotification
    )

    override fun equals(other: Any?): Boolean {
        if (this === other) return true // If the object is compared with itself then return true
        if (other !is Response) return false // If the object is not an instance of Response then return false
        return other.success == this.success && other.message == this.message
    }
}